src/Core/Collision/PolygonalCollider.cpp

```diff
@@ -356,12 +446,6 @@ namespace obe::Collision
     bool PolygonalCollider::doesCollide(
         PolygonalCollider& collider, const Transform::UnitVector& offset) const
     {
-        Transform::Rect boundingBox = this->getBoundingBox();
```

Comment:
Single `doesCollide` doesn't have bounding-box optimization ?

Reply:
The bounding-box optimization is already made in doesCollide(offset) and getMaximumDistanceBeforeCollision(offset). It avoids the process being made twice during collision detection.
An alternative would be to keep the optimization but add a boolean parameter to skip this process.

Reply:
Yup, I would keep the optimization, it's pretty useful when you want custom collision logic (such as climbing ladders etc..)